### PR TITLE
Add Facebook Test User API service and initial tests for Facebook flows

### PR DESCRIPTION
**Pull request by niklasvincent**
Makes it possible to programmatically create new Facebook users and delete them when a test is over.

Also uses the new service to test three different use cases. As part of that a few new page/step classes were added in.

This requires `facebookApplicationId` and `facebookApplicationSecret` being set with the Facebook application details in the `user` section in `local.conf`.

### CODE REVIEW

File: identity-tests/src/main/scala/com/gu/identity/integration/test/pages/FaceBookAuthDialog.scala [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+  private def emailCheckBox: WebElement = {
+    val xpath = "(//span[preceding::input[@value='email']])[1]"
+    val wait = new WebDriverWait(driver, 10);
+    wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath(xpath)))
+    driver.findElement(By.xpath(xpath))
+  }
+
+  def clickConfirmButton() = {
+    confirmButton.click()
```
Comment (jamesoram) [inline identity-tests/src/main/scala/com/gu/identity/integration/test/pages/FaceBookAuthDialog.scala:30]: we need to return a page here

File: identity-tests/src/main/scala/com/gu/identity/integration/test/pages/FacebookParentPage.scala [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+import com.gu.integration.test.pages.common.ParentPage
+import com.gu.integration.test.util.ElementLoader._
+import org.openqa.selenium.{By, WebDriver, WebElement}
+
+class FacebookParentPage(implicit driver: WebDriver) extends ParentPage {
+  private def emailInputField: WebElement = driver.findElement(By.id("email"))
+  private def passwordInputField: WebElement = driver.findElement(By.id("pass"))
+
+  def loginInButton: WebElement = driver.findElement(By.xpath("//input[@value='Log in']"))
```
Comment (jamesoram) [inline identity-tests/src/main/scala/com/gu/identity/integration/test/pages/FacebookParentPage.scala]: this should probably be private

File: identity-tests/src/test/scala/com/gu/identity/integration/test/features/SocialNetworkTests.scala [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+      val facebookUser = SocialNetworkSteps().createNewFacebookTestUser()
+      SocialNetworkSteps().goToFacebookAsUser(facebookUser)
+      BaseSteps().goToStartPage()
+      val registerPage = SignInSteps().clickSignInLink().clickRegisterNewUserLink()
+      val authDialog = registerPage.switchToNewSignIn().clickRegisterWithFacebookButton()
+      authDialog.clickConfirmButton()
+      UserSteps().clearLoginCookies()
+      BaseSteps().goToStartPage()
+      SocialNetworkSteps().checkUserGotAutoSignInBanner()
+      SignInSteps().checkUserIsLoggedIn(facebookUser.fullName)
```
Comment (jamesoram) [inline identity-tests/src/test/scala/com/gu/identity/integration/test/features/SocialNetworkTests.scala:52]: We could add a check for the GU_U cookie to exist

File: identity-tests/src/main/scala/com/gu/identity/integration/test/pages/FaceBookAuthDialog.scala (hunk 1 above)
Comment (niklasvincent) [inline identity-tests/src/main/scala/com/gu/identity/integration/test/pages/FaceBookAuthDialog.scala:30]: Now returns a page

File: identity-tests/src/test/scala/com/gu/identity/integration/test/steps/SocialNetworkSteps.scala [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    facebookSignInPage.enterPassword(facebookTestUser.password.get)
+    facebookSignInPage.clickLoginButton()
+
+    facebookSignInPage
+  }
+
+  // TODO(nlindblad): Should go into a page
+  def checkUserGotFacebookEmailError() = {
+    try {
+      val formError = driver.findElement(By.xpath("//div[@class='form__error']")).getText()
```
Comment (jamesoram) [inline identity-tests/src/test/scala/com/gu/identity/integration/test/steps/SocialNetworkSteps.scala]: It would be nice to clear up these TODOs :)

**Comment by jamesoram** [comment]
:+1: good work!

**Pull request merged**